src/MotorPlugin.cc

```diff
@@ -0,0 +1,704 @@
+
+			if (std::abs(pwm) < 1.0) 
+			{
+				// gzdbg << "Zero rad/s PWM from topic, setting torque to 0\n";
+				auto jfcComp = _ecm.Component<gz::sim::components::JointForceCmd>(control.joint);
+				if (jfcComp)
+				{
+					auto &forceCmd = jfcComp->Data();
+					if (!forceCmd.empty())
+					{
+						forceCmd[0] = 0.0;
+					}
+				}
+				continue;
+			}
+
+			double dt =std::chrono::duration_cast<std::chrono::duration<double> >(
+									_info.simTime - this->impl->
+					lastMotorModelUpdateTime).count();
+      gzdbg << "Dt:- " << dt << "\n";
+			double velError = targetOmega - currOmega;
+			double desOmega = this->impl->controls[i].pid.Update(
+					velError, std::chrono::duration<double>(dt)); 
+			
+
+			// Apply Drela's motor model equations
+			
+			// Equation (4): Current as function of speed and terminal voltage
+			// i(Ω, v) = (v - Ω/KV) / R
+			// Note: velocityConstant should be KV in rad/s/Volt
+
+			double reqV = (desOmega / control.velocityConstant) +   
+														(control.noLoadCurrent * control.coilResistance);
+			
+			// Clamp voltage to available range
+			double terminalV = std::max(-control.maxVolts, std::min(control.maxVolts, reqV));
+			
+			double backEmfV = currOmega / control.velocityConstant;  // Ω/KV
+			double current = (terminalV - backEmfV) / control.coilResistance;
+
+			// Equation (5): Qm = (i - io) / KQ
+			double torque = (current - control.noLoadCurrent) * control.velocityConstant;
+			
+			// Debug output
+			gzdbg << "Joint [" << control.jointName << "]:\n"
+						<< "  PWM: " << pwm << "\n"
+						<< "  Terminal Voltage: " << terminalV << " V\n"
+						<< "  Current Speed: " << currOmega << " rad/s (" << (currOmega * 60.0) / (2.0 * M_PI) << " RPM)\n"
+						<< "  Back-EMF: " << backEmfV << " V\n"
+						<< "  Current: " << current << " A\n"
+						<< "  Torque: " << torque << " Nm\n";
+
+			// Apply torque to joint
+			auto jfcComp = _ecm.Component<gz::sim::components::JointForceCmd>(control.joint);
+			if (jfcComp)
+			{
+				auto &forceCmd = jfcComp->Data();
+				forceCmd[0] = torque;
+			}
+			else
+			{
+				gzerr << "JointForceCmd component missing for joint [" << control.jointName << "]\n";
+			}
+		}
+
+    // Update parameters
+    this->impl->pGain.Update();
+    this->impl->iGain.Update();
+    this->impl->dGain.Update();
+    this->impl->iMax.Update();
+    this->impl->iMin.Update();
+    this->impl->cmdMax.Update();
+    this->impl->cmdMin.Update();
+    this->impl->cmdOffset.Update();
+	}
+
+}
+
+void MotorPlugin::PostUpdate(
+    const gz::sim::UpdateInfo &_info,
+    const gz::sim::EntityComponentManager &_ecm)
+{
+    std::lock_guard<std::mutex> lock(this->impl->pwmMutex);
+
+    // Publish the new state.
+    if (!_info.paused && _info.simTime > this->impl->lastMotorModelUpdateTime)
+    {
+        double t = std::chrono::duration_cast<std::chrono::duration<double>>(
+                _info.simTime).count();
+        this->impl->lastMotorModelUpdateTime = _info.simTime;
+    }
+}
```

Comment:
I think this can go? See

```c++
void JointController::PreUpdate(const UpdateInfo &_info,
    EntityComponentManager &_ecm)
```

which uses 

```c++
      double force = this->dataPtr->velPid.Update(error, _info.dt);
```

rather than maintain a `lastMotorModelUpdateTime`.